src/Controller/IslandoraCitationsController.php

```diff
@@ -76,11 +90,59 @@ public function provideArguments($node_type) {
     ];
   }
 
+  /**
+   * Helper; generate link to the field configuration page.
+   *
+   * @param string $type
+   *   The type of entity to which the field is associated.
+   * @param \Drupal\Core\Config\Entity\ConfigEntityBundleBase $bundle
+   *   The bundle with which the field is associated.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The field definition of which to link to the configuration page.
+   *
+   * @return \Drupal\Core\Link|\Drupal\Core\StringTranslation\TranslatableMarkup
+   *   A link to a page to configure the given field, or a string.
+   */
+  protected function getLinkToField(string $type, ConfigEntityBundleBase $bundle, FieldDefinitionInterface $field_definition) {
+    $add_destination = static function (Url $url) {
+      $query = $url->getOption('query');
+      $query['destination'] = Url::fromRoute('<current>')->toString();
+      $url->setOption('query', $query);
+      return $url;
+    };
+    /** @var \Drupal\Core\Field\Entity\BaseFieldOverride|\Drupal\field\Entity\FieldConfig $config */
+    $config = $field_definition->getConfig($bundle->id());
+    if ($config instanceof BaseFieldOverride) {
+      if ($this->moduleHandler()->moduleExists('base_field_override_ui')) {
+        return $config->isNew() ?
+          new Link(
+            $this->t('Add'),
+            $add_destination(BaseFieldOverrideUI::getAddRouteInfo($config)),
+          ) :
```

Comment:
An outdated comment is applicable to this section of code so linking for future reference https://github.com/discoverygarden/islandora_citations/pull/77#discussion_r1699093138

And yes I can confirm this is currently not applicable to the list of fields being iterated on, but I see no harm keeping it here in case things get extended/altered to iterate over all base fields rather than just the ones already included in a given content type.